gammapy/data/data_store.py

```diff
@@ -258,7 +266,15 @@ def get_observations(self, obs_id=None, skip_missing=False):
                 else:
                     raise err
             else:
-                obs_list.append(obs)
+                flag = True
+                for irf in required_IRF:
+                    if getattr(obs, irf) is None:
```

Comment:
I realised this implementation is problematic, because it will load existing IRFs from file, which makes it very inefficient. The problem is that once you call `getattr()` the `LazyFitsData` descriptor will load the data from disk. To avoid this you directly have to access hidden attributes. If you print the `Observation.__dict__` object:
```python3
{'obs_id': 23526,
 'aeff': None,
 '_edisp_hdu': <gammapy.utils.fits.HDULocation at 0x10601aee0>,
 '_psf_hdu': <gammapy.utils.fits.HDULocation at 0x1060b78e0>,
 '_bkg_hdu': <gammapy.utils.fits.HDULocation at 0x1060ab1c0>,
 '__gti_hdu': <gammapy.utils.fits.HDULocation at 0x1060b7070>,
 '__events_hdu': <gammapy.utils.fits.HDULocation at 0x1060b7a60>,
 'obs_filter': <gammapy.data.filters.ObservationFilter at 0x138f100d0>}
```
You can see, that missing irfs, are set to `None`, while the existing ones are defined as hidden data members with the `HDULocation` defining the location of the data. So you need work with e.g. `obs.__dict__.get(irf_name, False) is None` instead.